spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -885,6 +885,83 @@ would have all the data from those three sources sent to the file sink.
 The Fan-out use case is when you determine the destination of a stream based on some information that is only known at runtime.
 In this case, the link:http://docs.spring.io/spring-cloud-stream-app-starters/docs/Bacon.RELEASE/reference/html/spring-cloud-stream-modules-sinks.html#spring-cloud-stream-modules-router-sink[Router Application] can be used to specify how to direct the incoming message to one of N named destinations.
 
+[[spring-cloud-dataflow-stream-java-dsl]]
+== Stream Java DSL
+
+If you prefer to deploy streams programmatically, you can do so by using the fluent style DSL api provided by the `spring-cloud-dataflow-rest-client` module.
+The java DSL is a convenient wrapper around the `DataFlowTemplate` class that makes it simpler to create and deploy applications.
+Start by adding it to your project:
+
+[source,xml,subs="attributes+"]
+----
+<dependency>
+	<groupId>org.springframework.cloud</groupId>
+	<artifactId>spring-cloud-dataflow-rest-client</artifactId>
+	<version>{project-version}</version>
+</dependency>
+----
+
+To create a Stream you need to supply a `DataFlowTemplate` to the `Stream.Builder` static method:
+[source,java]
+----
+DataFlowTemplate client = //...;
```

Comment:
```		URI dataFlowUri = URI.create("http://localhost:9393");
		DataFlowOperations dataFlowOperations = new DataFlowTemplate(dataFlowUri);
```
Actually I remember we don't even print `9393` on startup...need to tweak the logging.